src/xdotoolify.js

```diff
@@ -12,24 +12,32 @@ const _waitForDOM = async function(page, timeout) {
 
   if (!(page && page.executeScript)) { return; }
 
-  return new Promise(async (resolve, reject) => {
-    const i = setInterval(async () => {
-      let readyState;
-
-      try {
-        readyState = await page.executeScript(function() {
-          return document.readyState;
-        });
-      } catch (e) {
-        // this error is thrown when a tab is closed
-        // so we ignore it
-        if (e.name === 'NoSuchWindowError') {
-          resolve();
-        } else {
-          throw e;
-        }
+  const getReadyState = async () => {
+    try {
+      const readyState = await page.executeScript(function() {
+        return document.readyState;
+      });
+      return readyState;
+    } catch (e) {
+      // this error is thrown when a tab is closed
+      // so we ignore it
+      if (e.name === 'NoSuchWindowError') {
+        return 'complete';
+      } else {
+        throw e;
       }
+    }
+  };
+
+  const readyState = await getReadyState();
```

Comment:
This is kind of awkardly written. This entire section can be replaced with a much simpler

```
return new Promise(async (resolve, reject) => {
  while (Date.now() < expires) {
    if (await getReadyState() === 'complete') {
      return resolve();
    }
    await new Promise((r) => setTimeout(r, 20)); // can be replaced with await sleep(20) if you've defined it
  }
  reject('Timed out while waiting for the dom to load');
})
```

Reply:
Replaced.